android/build.gradle

```diff
@@ -107,8 +107,8 @@ afterEvaluate { project ->
 
     android.libraryVariants.all { variant ->
         def name = variant.name.capitalize()
-        task "jar${name}"(type: Jar, dependsOn: variant.javaCompile) {
-            from variant.javaCompile.destinationDir
+        task "jar${name}"(type: Jar, dependsOn: variant.javaCompileProvider) {
+            from variant.javaCompileProvider.get().destinationDir
```

Comment:
Not quite familiar with react native. But when I apply this commit to my project, got the below error.

```
> Configure project :react-native-securerandom
WARNING:Software Components will not be created automatically for Maven publishing from Android Gradle Plugin 8.0. To opt-in to the future behavior, set the Gradle property android.disableAutomaticComponentCreation=true in the `gradle.properties` file or use the new publishing DSL.

FAILURE: Build failed with an exception.

* Where:
Build file '/Volumes/workplace/test_5930242856/src/MShopSsnapSharedComponentsAndroid/node_modules/react-native-securerandom/android/build.gradle' line: 111

* What went wrong:
A problem occurred configuring project ':react-native-securerandom'.
> Could not get unknown property 'destinationDir' for provider(task 'compileDebugJavaWithJavac', class org.gradle.api.tasks.compile.JavaCompile) of type org.gradle.api.internal.tasks.DefaultTaskContainer$TaskCreatingProvider_Decorated.
```

Remove changes on line 110~111 worked for me.